infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -405,29 +391,42 @@ private void registerToLogReplicationLock() {
     }
 
     /**
-     * This method is only called on the leader node and it triggers the start of log replication
+     * This method is only called on the leader node and is triggered on the start of log replication and on topology change
      * <p>
-     * Depending on the role of the cluster to which this leader node belongs, it will start
-     * as source (sender/producer) or sink (receiver).
+     * Depending on the role of the cluster to which this leader node belongs (Source, Sink or both), it will create
+     * respective components, and trigger connection if local cluster is the connection starter to any remote cluster,
+     * else we wait to receive a connection request.
+     * Replication starts after connection to/from a remote cluster is successful.
      */
     private void onLeadershipAcquire() {
-        // TODO [V2]: a cluster can be Source and/or Sink for different replication models. The support for this is
-        //  coming in the connectionModel PR
-        if (isSource()) {
-            log.info("Start as Source (sender/replicator)");
-            sessionManager.startReplication();
-            lockAcquireSample = recordLockAcquire();
-            processCountOnLockAcquire();
-        } else if (isSink()) {
-            // TODO: Shama: in connectionModel PR, log session info when cluster is a sink
-            log.info("Start as Sink (receiver)");
-            // Sink Site : the LogReplicationServer (server handler) will
-            // reset the LogReplicationSinkManager on acquiring leadership
-            interClusterServerNode.setLeadership(true);
-            lockAcquireSample = recordLockAcquire();
-            processCountOnLockAcquire();
-        } else {
+        if (!isSource() && !isSink()) {
             log.error("Log Replication not started on this cluster. Remote source/sink not found");
+            return;
+        }
+        setupConnectionComponents();
+        // record metrics about the lock acquired.
+        lockAcquireSample = recordLockAcquire();
+        processCountOnLockAcquire();
+    }
+
+    private void setupConnectionComponents() {
+        setupConnectionReceivingComponents();
+        sessionManager.connectToRemoteClusters();
+    }
+
+    /**
+     * Create CorfuInterClusterReplicationServerNode when the cluster is a connection endpoint.
+     */
+    private void setupConnectionReceivingComponents() {
+        if (!sessionManager.isConnectionReceiver()) {
+            return;
```

Comment:
I think we can skip having a log statement here, because while debugging, we would look if remote cluster is starting the connection to someone or if we are starting the server...there are logs down the lane for both these.
In the absence of both, we'll then look at topology to check why the local cluster is neither connection starter or connection receiver.

So the log statement here would be redundant.